src/nvim/os/keyboard.c

```diff
@@ -0,0 +1,295 @@
+// This is an open source non-commercial project. Dear PVS-Studio, please check
+// it. PVS-Studio Static Code Analyzer for C, C++ and C#: http://www.viva64.com
+
+#include <assert.h>
+#include <string.h>
+#include <stdio.h>
+
+#include <uv.h>
+
+#include "nvim/api/private/helpers.h"
+#include "nvim/api/private/defs.h"
+#include "nvim/lua/executor.h"
+#include "nvim/vim.h"
+#include "nvim/version.h"
+
+#ifdef __linux__
+
+#define XLIB_NAME "libX11.so.6"
+typedef void *_XDisplay;
+typedef struct {
+  char data[18];
+} _XkbStateRec;
+typedef int _XStatus;
+
+///
+enum { kLockedModsOffset = 9 };
+enum { _XkbUseCoreKbd = 0x0100 };
+
+/// Masks used in libX11 for different Lock states
+typedef enum {
+    xNumLock = 0x10,  ///< Mask for NumLock
+    xCapsLock = 0x02,  ///< Mask for CapsLock
+    xScrollLock = 0x80,  ///< Mask for ScrollLock
+} xlib_KbdLocks;
+
+#endif
+
+#ifdef WIN32
+
+#include <Winuser.h>
+
+#endif
+
+#ifdef __APPLE__
+
+#define Boolean Boolean_I_Dont_Care
```

Comment:
These should be indented, one space between hash and text. Also I would really suggest to get rid of newlines below and above.

Reply:
Is the indentation is required for `include`s too, some of the files in neovim code use indentation for `include`s but some code don't use it, so it was ambiguous to me as earlier I used no space.

I add space, if the coding style says